iceoryx_examples/automotive_soa/iox_automotive_proxy.cpp

```diff
@@ -0,0 +1,188 @@
+                          << MinimalProxy::m_serviceIdentifier << "' has disappeared." << std::endl;
+                maybeProxy->value().m_event.UnsetReceiveCallback();
+                maybeProxy->reset();
+                return;
+            }
+            //! [destroy proxy asynchronously]
+
+            //! [create proxy asynchronously]
+            for (auto& proxyHandle : container)
+            {
+                if (!maybeProxy->has_value())
+                {
+                    std::cout << "  Found instance of service: '" << MinimalProxy::m_serviceIdentifier << "', '"
+                              << proxyHandle.GetInstanceId().c_str() << "'" << std::endl;
+                    maybeProxy->emplace(proxyHandle);
+                    break;
+                }
+            }
+            //! [create proxy asynchronously]
+        };
+
+        //! [set up asynchronous search]
+        auto handle = MinimalProxy::EnableFindServiceCallback(callback, exampleInstanceSearchQuery);
+        //! [set up asynchronous search]
+        maybeHandle.emplace(handle);
+        std::cout << "  Waiting for instance called '" << exampleInstanceSearchQuery.c_str()
+                  << "' to become available.." << std::endl;
+    }
+    //! [Field: create ints for computeSum]
+    uint64_t addend1{0};
+    uint64_t addend2{0};
+    //! [Field: create ints for computeSum]
+
+    while (!iox::posix::hasTerminationRequested())
+    {
+        {
+            //! [gain exclusive access to proxy]
+            auto proxyGuard = maybeProxy.getScopeGuard();
+            if (proxyGuard->has_value())
+            {
+                auto& proxy = proxyGuard->value();
+                //! [gain exclusive access to proxy]
+
+                //! [Event: receiveCallback]
+                auto onReceive = [&]() -> void {
+                    proxy.m_event.TakeNewSamples([](const auto& topic) {
+                        auto finish = std::chrono::steady_clock::now();
+                        std::cout << "Event: value is " << topic->counter << std::endl;
+                        auto duration =
+                            std::chrono::duration_cast<std::chrono::nanoseconds>(finish - topic->sendTimestamp);
+                        std::cout << "Event: latency (ns) is " << duration.count() << std::endl;
+                    });
+                };
+                //! [Event: receiveCallback]
+
+                //! [Event: set receiveCallback]
+                if (!proxy.m_event.HasReceiveCallback())
+                {
+                    proxy.m_event.Subscribe(10U);
+                    proxy.m_event.SetReceiveCallback(onReceive);
+                }
+                //! [Event: set receiveCallback]
+
+                //! [Field: get and set data]
+                auto fieldFuture = proxy.m_field.Get();
+                try
+                {
+                    auto result = fieldFuture.get();
```

Comment:
Instead of accessing the future here directly and risk an exception shouldn't we call `fieldFuture.valid()` to check if it contains something and then go on with the code.
It would have the benefit of never throwing an exception and it would be also according to our idioms.

Reply:
@elfenpiff This is correct. Here, this is an explicit choice to not follow the iceoryx idioms and use a `try`-`catch` to make developers aware that this is different to the iceoryx code. In the future (pun intended!) we might have a `cxx::future` implementation from @MatthiasKillat which uses our iceoryx idioms. Then I would be ok with changing this.